File: C++/findPeakElement.cpp
#include<iostream>
#include<vector>

class Solution{
    public:
    int findPeakElement(std::vector<int>& nums) {
        size_t begin = 0;
        size_t end = nums.size()-1;
        while(begin<end){
            size_t med =begin+((end-begin)>>1);
            if(nums[med]>nums[med+1]){
                end=med;
            }
            else{
                begin=med+1;
            }
        }
        return begin;
    }
};

int main(){
    std::vector<int>nums={2,4,1,2,7,8,4};
    Solution solution;
    int PeakElement=solution.findPeakElement(nums);
    std::cout<<PeakElement<<std::endl;
    return 0;
}

File: python/findPeakElement.py
class Solution:
    def findPeakElement(self , nums: list[int]) -> int:
        begin = 0
        end = len(nums) - 1
        
        while begin < end:
            med = begin + (end - begin) // 2
            if nums[med] < nums[med + 1]:
                begin = med + 1
            else:
                end = med
        
        return begin
    
nums=[2,4,1,2,7,8,4]
solution=Solution()
print(solution.findPeakElement(nums))